osquery/dispatcher/scheduler.cpp

```diff
@@ -150,38 +151,64 @@ inline void launchQuery(const std::string& name, const ScheduledQuery& query) {
 }
 
 void SchedulerRunner::start() {
-  // Start the counter at the second.
-  auto i = osquery::getUnixTime();
-  for (; (timeout_ == 0) || (i <= timeout_); ++i) {
-    Config::get().scheduledQueries(
-        ([&i](const std::string& name, const ScheduledQuery& query) {
-          if (query.splayed_interval > 0 && i % query.splayed_interval == 0) {
-            TablePlugin::kCacheInterval = query.splayed_interval;
-            TablePlugin::kCacheStep = i;
-            launchQuery(name, query);
-          }
-        }));
+  auto tasks = ScheduledTaskQueue{};
+  Config::get().scheduledQueries(
+      [&tasks](const std::string& name, const ScheduledQuery& query) {
+        if (query.splayed_interval > 0) {
+          tasks.add(
+              [name, &query](auto startTime) {
```

Comment:
Why is name not captured as a reference?

startTime -> start_time
Should startTime be const?

Reply:
`name` is captured by value just because the name reference passed to the first lambda is located on stack. And when we call the task implementation lambda the reals string with name does not exist. So we need to save the copy of that string inside the lambda. To fix this confusing in the future I have made another [PR4367](https://github.com/facebook/osquery/pull/4367).

`'s/startTime/start_time/'` - done

`start_time` - it can be const, but it is exact match of interface from ScheduledTaskQueue::add.